rero_ils/alembic/ce4923ba5286_delete_expired_items_temporary_locations.py

```diff
@@ -0,0 +1,52 @@
+# -*- coding: utf-8 -*-
+#
+# RERO ILS
+# Copyright (C) 2022 RERO
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published by
+# the Free Software Foundation, version 3 of the License.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+"""Delete expired items temporary locations."""
+
+from logging import getLogger
+
+from rero_ils.modules.items.api import Item, ItemsSearch
+
+revision = 'ce4923ba5286'
+down_revision = '05555c03fe49'
+branch_labels = ()
+depends_on = None
+
+LOGGER = getLogger('alembic')
+
+
+def index_items_with_temporary_location():
+    """Index items with temporary location."""
+    query = ItemsSearch() \
+        .filter('exists', field='temporary_location') \
+        .source(['pid']).scan()
+    for hit in query:
+        item = Item.get_record_by_id(hit.meta.id)
+        item.reindex()
+        LOGGER.info(f'  * Reindexed item#{hit.pid}')
```

Comment:
1) You should get the record uuid from ES directly. It avoids to get the record (DB access, ...) is not needed.
2) Try to index more than 1 record using a bulk indexing (much more faster).

Something like 
```python
chunck_size = 1000
search = ItemsSearch().filter('exists', field='temporary_location')
uuids = [hit.meta.id for hit in search.scan()]
chunks = [uuids[x:x + chunck_size] for x in range(0, len(uuids), chunck_size)]
indexer = ItemsIndexer()
for chunk in chunks:
    indexer.bulk_index(uuids)
    count =  indexer.process_bulk_queue()
    LOGGER.info(f'{count} records indexed.')
```